### PR TITLE
Add retries to tasks that commonly fail in CI

### PR DESCRIPTION
**Pull request by jovial**
These tasks have been seen to fail:

- stackhpc.pulp.pulp_repository : Sync container remotes into repositories

- stackhpc.pulp.pulp_repository : Setup container remotes]

I added repository retries for consistency.

### CODE REVIEW

File: roles/pulp_repository/tasks/container.yml [hunk 1]
```diff
@@ -10,6 +10,10 @@
   loop: "{{ pulp_repository_container_repos | map(attribute='name') }}"
   loop_control:
     index_var: repository_index
+  register: pulp_repository_container_repositories
+  until: "pulp_repository_container_repositories is not failed"
+  retries: "{{ pulp_repository_container_repositories_retries }}"
```
Comment (jovial) [inline roles/pulp_repository/tasks/container.yml:15]:
```suggestion
  retries: "{{ pulp_repository_container_repo_repositories_retries }}"
```

File: roles/pulp_repository/tasks/container.yml [hunk 2]
```diff
@@ -40,6 +44,10 @@
   loop: "{{ pulp_repository_container_repos | map(attribute='name') }}"
   loop_control:
     index_var: repository_index
+  register: pulp_repository_container_remotes
+  until: "pulp_repository_container_remotes is not failed"
+  retries: "{{ pulp_repository_container_remotes_retries }}"
```
Comment (jovial) [inline roles/pulp_repository/tasks/container.yml:49]:
```suggestion
  retries: "{{ pulp_repository_container_repos_remotes_retries }}"
```

File: roles/pulp_repository/defaults/main.yml [hunk 3]
```diff
@@ -12,6 +12,14 @@ pulp_repository_rpm_repos: []
...
+pulp_repository_container_repos_repositories_retries: "{{ pulp_repository_sync_retries }}"
 pulp_repository_deb_repos_sync_retries: "{{ pulp_repository_sync_retries }}"
+pulp_repository_deb_repos_remotes_retries: "{{ pulp_repository_sync_retries }}"
+pulp_repository_deb_repos_repositories_retries: "{{ pulp_repository_sync_retries }}"
 pulp_repository_python_repos_sync_retries: "{{ pulp_repository_sync_retries }}"
+pulp_repository_python_repos_remotes_retries: "{{ pulp_repository_sync_retries }}"
+pulp_repository_python_repos_repositories_retries: "{{ pulp_repository_sync_retries }}"
 pulp_repository_rpm_repos_sync_retries: "{{ pulp_repository_sync_retries }}"
+pulp_repository_rpm_repos_remotes_retries: "{{ pulp_repository_sync_retries }}"
+pulp_repository_rpm_repos_repositories_retries: "{{ pulp_repository_sync_retries }}"
```
Comment (jovial) [inline roles/pulp_repository/defaults/main.yml:25]: Will remove the repos_ prefix

Comment (jovial) [reply roles/pulp_repository/defaults/main.yml]: > Will remove the repos_ prefix

In two minds, in some ways the repos_ prefix is more consistent. Let me know what you think.

File: roles/pulp_repository/defaults/main.yml [hunk 4]
```diff
@@ -12,6 +12,14 @@ pulp_repository_rpm_repos: []
 pulp_repository_sync_retries: 3
```
Comment (markgoddard) [inline roles/pulp_repository/defaults/main.yml:12]: Naming is a little off here. Would suggest either renaming this to `pulp_repository_retries` or adding separate defaults for repositories and remotes.

Comment (jovial) [reply roles/pulp_repository/defaults/main.yml:12]: Good call

**Review by markgoddard — APPROVED** [review]

**Comment by markgoddard** [comment]
Released in v0.5.3: https://github.com/stackhpc/ansible-collection-pulp/releases/tag/v0.5.3

**Comment by markgoddard** [comment]
Will need an update to stackhpc-release-train to pull it in.

**Pull request merged**